web/src/settings_components.ts

```diff
@@ -406,68 +515,93 @@ export function change_save_button_state($element, state) {
             break;
     }
 
+    assert(button_text !== undefined);
     $textEl.text(button_text);
+    assert(data_status !== undefined);
     $saveBtn.attr("data-status", data_status);
     if (state === "unsaved") {
         enable_or_disable_save_button($element.closest(".settings-subsection-parent"));
     }
-    show_hide_element($element, is_show, 800);
+    assert(is_show !== undefined);
+    show_hide_element($element, is_show, 800, () => {
+        // There is no need for a callback here since we have already
+        // called the function to enable or disable save button.
+    });
 }
 
-function get_input_type($input_elem, input_type) {
+function get_input_type($input_elem: JQuery, input_type: string): string {
     if (["boolean", "string", "number"].includes(input_type)) {
         return input_type;
     }
     return $input_elem.data("setting-widget-type");
 }
 
-export function get_input_element_value(input_elem, input_type) {
+export function get_input_element_value(
+    input_elem: HTMLElement,
+    input_type: string,
+): number | string | null | undefined {
     const $input_elem = $(input_elem);
     input_type = get_input_type($input_elem, input_type);
+    let input_value;
     switch (input_type) {
         case "boolean":
-            return $input_elem.prop("checked");
+            return $(input_elem).prop("checked");
         case "string":
-            return $input_elem.val().trim();
+            assert(
+                input_elem instanceof HTMLInputElement || input_elem instanceof HTMLSelectElement,
+            );
+            input_value = $(input_elem).val()!;
+            assert(typeof input_value === "string");
```

Comment:
Cannot assert the type to be `HTMLSelectElement & {type: "select-one"}` and that's why added the `typeof` assert to make sure it is just a string and not string array.

Reply:
It seems like a TypeScript bug that `assert(input_elem instanceof HTMLSelectElement && input_elem.type === "select-one")` doesn’t suffice. microsoft/TypeScript-DOM-lib-generator#1697 might fix it.

Perhaps the best solution for now is to carefully define a user-defined type guard for the `HTMLSelectElement & {type: "select-one"}` type.

Reply:
I wanted to confirm whether the solution that I have thought of based on my understanding of user defined type guards 
works or not -
```js
function is_single_select_elem(
    elem: HTMLElement,
): elem is HTMLSelectElement & {type: "select-one"} {
    return elem instanceof HTMLSelectElement && elem.type === "select-one";
}

export function get_input_element_value(
    input_elem: HTMLElement,
    input_type?: string,
): number | string | null | undefined {
    const $input_elem = $(input_elem);
    input_type = get_input_type($input_elem, input_type);
    let input_value;
    switch (input_type) {
        case "boolean":
            return $(input_elem).prop("checked");
        .......
        .......
        case "jitsi-server-url-setting":
            if (is_single_select_elem(input_elem)) {
                return get_jitsi_server_url_setting_value(input_elem);
            }
            return 0;
       .......
       .......
```

I had to change `get_jitsi_server_url_setting_value` to accept html element instead of jquery object to make it work because `is_single_select_elem` needs HTML element.